File: examples/matlab/request_reply/Reply.m
function Reply(mm_ip)

    global RTMA;

    RTMA_BaseDir = '../../..';
    addpath([RTMA_BaseDir '/lang/matlab']);

    RTMA_BaseDir

    MessageTypes =  { ...
                        'REQUEST_TEST_DATA' ...
                        'EXIT' ...
                    };

    ConnectArgs = {'REPLY', RTMA_BaseDir, ['./message_defs.mat']};
    if exist('mm_ip','var') && ~isempty(mm_ip)
        ConnectArgs{end+1} = ['-server_name ' mm_ip];
    end

    ConnectToMMM(ConnectArgs{:});
    Subscribe( MessageTypes{:})
    
    disp 'Reply running..'

    cnt = 1;
    while( 1)
        fprintf('\nWaiting for message\n');
        M = ReadMessage( 'blocking');
        
        switch(M.msg_type)
            case 'REQUEST_TEST_DATA'
                fprintf('Received message %s\n', M.msg_type);
            
                msg = RTMA.MDF.TEST_DATA;
                msg.a = int32(cnt);
                msg.b = int32(-30);
                msg.x = 123.456;
                SendMessage( 'TEST_DATA', msg);
                
                fprintf('Sent reply\n');
                cnt = cnt + 1;
            case 'EXIT'
                break; 

        end

    end

    DisconnectFromMMM

File: examples/python/request_reply/Reply.py
#!/usr/bin/python
from __future__ import print_function
import time
import PyRTMA
from PyRTMA import copy_from_msg, copy_to_msg, MT_EXIT, MT_KILL
import message_defs as mdefs
import sys

MID_REPLY = 10

# Note: Reply must be started first

if __name__ == "__main__":
    mod = PyRTMA.RTMA_Module(MID_REPLY, 0)
    mod.ConnectToMMM()
    mod.Subscribe(mdefs.MT_REQUEST_TEST_DATA)
    
    print("Reply running...\n")
    
    run = True
    while run:
        in_msg = PyRTMA.CMessage()
        print("Waiting for message")
        rcv = mod.ReadMessage(in_msg, 1.0)
        if rcv == 1:
            print("Received message", in_msg.GetHeader().msg_type)
            out_msg = PyRTMA.CMessage(mdefs.MT_TEST_DATA)
            data = mdefs.MDF_TEST_DATA()
            if in_msg.GetHeader().msg_type == mdefs.MT_REQUEST_TEST_DATA:
                data.a = -20
                data.b = 47
                data.x = 123.456
                copy_to_msg(data, out_msg)
                mod.SendMessage(out_msg)
                print("Sent message", out_msg.GetHeader().msg_type)
            elif in_msg.GetHeader().msg_type in (MT_EXIT, MT_KILL):
               run = False        
